src/transformers/models/llama/modeling_llama.py

```diff
@@ -1148,6 +1178,10 @@ def forward(
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
         cache_position: Optional[torch.LongTensor] = None,
+        cu_seq_lens_q: Optional[torch.LongTensor] = None,
+        cu_seq_lens_k: Optional[torch.LongTensor] = None,
+        max_length_q: int = 0,
+        max_length_k: int = 0,
```

Comment:
these are are FlashAttention specific. IMO it would make sense to just add them as `fa2_kwargs` for example. We can use something like this: https://github.com/huggingface/transformers/blob/ce81eb8d14ae39e183a8956cce3f7e2e46ecaa8f/src/transformers/processing_utils.py#L81